<launch>

    <arg name="rgb_camera" default="/camera/color/"/>
    <arg name="rgb_camera_topic" default="$(arg rgb_camera)/image_raw"/>
    <arg name="rgb_camera_info" default="$(arg rgb_camera)/camera_info"/>

    <arg name="use_depth" default="False"/>
    <arg name="depth_camera" default="/camera/depth/"/>
    <arg name="depth_camera_topic" default="$(arg depth_camera)/image_rect_raw"/>
    <arg name="depth_camera_info" default="$(arg depth_camera)/camera_info"/>

    <!-- publishes an image with the body bounding boxes -->
    <arg name="visual_debug" default="False"/>

    <!-- console output of detected bounding boxes -->
    <arg name="textual_debug" default="False"/>

    <!-- publishes a TF tree of the joints, *without* performing IK kinematics to
         properly align the joints' frames to the kinematci model of the human -->
    <arg name="stickman_debug" default="False"/>
    <arg name="single_body" default="False"/>


    <node name="hri_fullbody" pkg="hri_fullbody" type="detect" output="screen">
        <param name="use_depth" value="$(arg use_depth)"/>
        <param name="visual_debug" value="$(arg visual_debug)"/>
        <param name="textual_debug" value="$(arg textual_debug)"/>
        <param name="stickman_debug" value="$(arg stickman_debug)"/>
        <param name="single_body" value="$(arg single_body)"/>

        <remap from="/image" to="$(arg rgb_camera_topic)"/>
        <remap from="/image_info" to="$(arg rgb_camera_info)"/>
        <remap from="/depth_image" to="$(arg depth_camera_topic)"/>
        <remap from="/depth_info" to="$(arg depth_camera_info)"/>
    </node>

</launch>
